### ISSUE TITLE
Metrics plateau in YOLOv8 (help in model evaluation)

### ISSUE THREAD

**Issue by KDLPro** [ultralytics/ultralytics#16650]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello guys! I'm training an abaca fiber detection model. It has only one class, as per my thesis adviser's suggestion. Here are the details of the dataset:
![image](https://github.com/user-attachments/assets/b693b2fe-c35e-4390-8561-363c1d1e948e)
As for model training, the dataset is trained on Ultralytics YOLOv8 locally on 100 epochs with no additional parameters set (uses the default parameters). Here are the results of the training:
![image](https://github.com/user-attachments/assets/89ec39c0-47ca-4bad-8db4-beb9efb21079)
![image](https://github.com/user-attachments/assets/67e0f96c-7332-4fd0-aae6-1589a8ba29fb)
Why is the results looking like this where there are no huge changes on the precision and recall? Is there something wrong with the dataset or training? Your help is very much appreciated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @KDLPro, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will assist you soon.

For now, we suggest checking out our [Docs](https://docs.ultralytics.com) for guidance on model training and evaluation. You might find our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) helpful.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us understand the issue better.

For your inquiry about metrics plateauing, please ensure that your environment is set up with the latest software. You can update to the latest version of the `ultralytics` package, along with all its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a Python environment:

```bash
pip install -U ultralytics
```

Here are some community spaces where you can engage with other users and our team:
- For real-time chat, join our [Discord](https://ultralytics.com/discord) 🎧.
- For discussions, visit [Discourse](https://community.ultralytics.com).
- You can also check our [Subreddit](https://reddit.com/r/ultralytics) for more insights.

## Environments

YOLO is supported in multiple environments. You may try running it in the following:
- **Notebooks** with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Make sure to check the CI badge below to verify all tests are passing, ensuring the latest code operations are validated:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
Provide the training code you used.

**Comment by KDLPro** [comment]
@Y-T-G , what do you mean training code? I just installed Ultralytics on my computer (on Anaconda). I didn't make any changes to the training code.

**Comment by Y-T-G** [comment]
What command or code did you use to run training?

**Comment by KDLPro** [comment]
Here are the arguments used for training. Most of them were defaults.

```
task: detect
mode: train
model: yolov8s.yaml
data: data.yaml
epochs: 100
time: null
patience: 100
batch: 16
imgsz: 640
save: true
save_period: -1
cache: false
device: null
workers: 8
project: null
name: train8
exist_ok: false
pretrained: true
optimizer: auto
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
multi_scale: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: true
opset: null
workspace: 4
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
auto_augment: randaugment
erasing: 0.4
crop_fraction: 1.0
cfg: null
tracker: botsort.yaml
save_dir: runs\detect\train8
```

**Comment by Y-T-G** [comment]
I need to see the training command or code that you used to launch training. Showing all the arguments makes it difficult to know what you changed.

**Comment by Y-T-G** [comment]
From the screenshot, it seems like most of your images look similar. That's bad. There's no point in providing similar images in training dataset. That will make the model overfit. Even worse if you have the same similar images in the validation set, which would mean your model is simply validating on images that it has already seen or learnt.

It's better to have 100 unique images than a 1000 similar images.

**Comment by Y-T-G** [comment]
Follow the guidelines here:

https://docs.ultralytics.com/yolov5/tutorials/tips_for_best_training_results/

**Comment by KDLPro** [comment]
Oh I see. Abaca fiber grades have a quality standard that's why they are similar. So that means adding another class (with different images, not too similar to this one) will yield better results?

**Comment by Y-T-G** [comment]
You should remove similar images and only have distinct images. And make sure the images in validation are distinct from training. They shouldn't be just different angles of the same image.

**Comment by KDLPro** [comment]
I see. I was extracting a video in 20 frames per second to account for minute variations of angles (and since my classmate told me to move the camera slowly while recording, to increase the number of images in the dataset).

**Comment by KDLPro** [comment]
Also, lighting can play a role in classifying abaca fibers. With that said, is it okay to gather data using a different camera? The images were took on my friend's Canon EOS M3. My friend is back in his hometown though so I only have my phone camera on hand.

**Comment by Y-T-G** [comment]
You need data that looks distinct, and images with variations that are significant. Images that look similar don't provide any additional information to the model. The model can learn the same information just from one of them.

**Comment by KDLPro** [comment]
Here are some examples of abaca fibers per grade. Since abaca fibers of the same grade have little to no variations on one another, is it much better to train a dataset on multiple grades of fibers at once?
![image](https://github.com/user-attachments/assets/51922e15-4e5f-4bc1-9550-543cdd4951be)

**Comment by Y-T-G** [comment]
How many classes do you have? You should be training on images from all the classes.

**Comment by KDLPro** [comment]
My plan is 6 classes but my thesis adviser said to train the dataset on only one class at a time to make it easier for me.

**Comment by Y-T-G** [comment]
So one model for each?

It's better to train on all the classes in a single model so that the model learns to differentiate.

**Comment by KDLPro** [comment]
I mean one at a time. Planning to present this paper to a conference or a published paper. But she advised me for a submission to focus on detection of fibers.

**Comment by Y-T-G** [comment]
> I mean one at a time.

You can't train it like that. It will forget what it learnt previously while training the next one.

**Comment by KDLPro** [comment]
Ahhh I see, gotcha. Thanks for the insight!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by KDLPro** [comment]
Sure thing @glenn-jocher. Thanks for the help also @Y-T-G. Also, what does this specific training and validation losses mean?

**Comment by Y-T-G** [comment]
@KDLPro It shows overfitting.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by KDLPro** [comment]
> @KDLPro It shows overfitting.

Gotcha, thank you very much!

**Comment by KDLPro** [comment]
> @KDLPro It shows overfitting.

Gotcha, thank you very much!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by glenn-jocher** [comment]
@KDLPro you're welcome! If you have any more questions or need further assistance, feel free to ask.
